### ISSUE TITLE
wrong `no-unused-expression` detection

### ISSUE THREAD

**Issue by Ardeshir81** [palantir/tslint#4675]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.2.4
- __Running TSLint via__: (pick one)  VSCode 

#### TypeScript code being linted

```ts
ngOnInit() {
    !this.authService.isAdminUser() && this.authService.logOut();
  }
```

with `tslint.json` configuration:

```json
{
  "extends": "../tslint.json",
  "rules": {
    "directive-selector": [
      true,
      "attribute",
      "mrkz",
      "camelCase"
    ],
    "component-selector": [
      true,
      "element",
      "mrkz",
      "kebab-case"
    ]
  }
}

```
```json
{
  "extends": "tslint:recommended",
  "rulesDirectory": [
    "codelyzer"
  ],
  "rules": {
    "array-type": false,
    "arrow-parens": false,
    "deprecation": {
      "severity": "warn"
    },
    "import-blacklist": [
      true,
      "rxjs/Rx"
    ],
    "interface-name": false,
    "max-classes-per-file": false,
    "max-line-length": [
      true,
      140
    ],
    "member-access": false,
    "member-ordering": [
      true,
      {
        "order": [
          "static-field",
          "instance-field",
          "static-method",
          "instance-method"
        ]
      }
    ],
    "no-consecutive-blank-lines": false,
    "no-console": [
      true,
      "debug",
      "info",
      "time",
      "timeEnd",
      "trace"
    ],
    "no-empty": false,
    "no-inferrable-types": [
      true,
      "ignore-params"
    ],
    "no-non-null-assertion": true,
    "no-redundant-jsdoc": true,
    "no-switch-case-fall-through": true,
    "no-use-before-declare": true,
    "no-var-requires": false,
    "object-literal-key-quotes": [
      true,
      "as-needed"
    ],
    "object-literal-sort-keys": false,
    "ordered-imports": false,
    "quotemark": [
      true,
      "single"
    ],
    "trailing-comma": false,
    "no-output-on-prefix": true,
    "use-input-property-decorator": true,
    "use-output-property-decorator": true,
    "use-host-property-decorator": true,
    "no-input-rename": true,
    "no-output-rename": true,
    "use-life-cycle-interface": true,
    "use-pipe-transform-interface": true,
    "component-class-suffix": true,
    "directive-class-suffix": true
  }
}

```
(I added both `tslint`s from angular)

#### Actual behavior
I run a function conditionally, but `tslint` states that this expression must be an assignment or function call. But it is a function call

#### Expected behavior
correct detection 
![image](https://user-images.githubusercontent.com/5755214/56469258-88fbb480-644c-11e9-8313-7d5dab2c9aa7.png)

**Comment by adidahiya** [comment]
duplicate of #4021
